Sources/AsyncHTTPClient/HTTPHandler.swift

```diff
@@ -405,8 +405,8 @@ extension HTTPClient {
 
         @discardableResult
         func setChannel(_ channel: Channel) -> Channel {
-            precondition(self.eventLoop === channel.eventLoop, "Channel must use same event loop as this task.")
```

Comment:
AFAICT this isn't needed anymore now that the back-pressure futures are always hopped.